src/main/java/com/lunark/lunark/reservations/controller/ReservationController.java

```diff
@@ -15,10 +23,24 @@
 @RestController
 @RequestMapping("/api/reservations")
 public class ReservationController {
+    private final IReservationService reservationService;
+    private final ModelMapper modelMapper;
+
+    @Autowired
+    public ReservationController(IReservationService reservationService, ModelMapper modelMapper) {
+        this.reservationService = reservationService;
+        this.modelMapper = modelMapper;
+    }
+
     @PostMapping(path = "", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<ReservationConfirmationDto> createReservation(@RequestHeader("x-access-token") String guestToken, @RequestBody ReservationRequestDTO reservationRequestDTO) {
-        ReservationConfirmationDto reservationConfirmationDto = new ReservationConfirmationDto("Vila Golija", LocalDate.of(2024, 6, 14), LocalDate.of(2023, 6, 16), 9959, 3);
-        return new ResponseEntity<>(reservationConfirmationDto, HttpStatus.CREATED);
+    @PreAuthorize("hasAuthority('guest')")
```

Comment:
Did you mean to put this in uppercase? I get 403 errors if it's left as is.
```suggestion
    @PreAuthorize("hasAuthority('GUEST')")
```